macros/schema_tests/table_shape/expect_table_columns_not_to_contain_set.sql

```diff
@@ -0,0 +1,29 @@
+{%- test expect_table_columns_not_to_contain_set(model, column_list, transform="upper") -%}
+{%- if execute -%}
+    {%- set column_list = column_list | map(transform) | list -%}
+    {%- set relation_column_names = dbt_expectations._get_column_list(model, transform) -%}
+    {%- set matching_columns = dbt_expectations._list_intersect(column_list, relation_column_names) -%}
+    with relation_columns as (
+
+        {% for col_name in relation_column_names %}
+        select cast('{{ col_name }}' as {{ dbt_utils.type_string() }}) as relation_column
+        {% if not loop.last %}union all{% endif %}
+        {% endfor %}
+    ),
+    input_columns as (
+
+        {% for col_name in column_list %}
+        select cast('{{ col_name }}' as {{ dbt_utils.type_string() }}) as input_column
+        {% if not loop.last %}union all{% endif %}
+        {% endfor %}
+    )
+    select *
+    from
+        input_columns i
+        left join
+        relation_columns r on r.relation_column = i.input_column
+    where
+        -- catch any column in input list that is not in the list of table columns
```

Comment:
I think the comment should read:
```
-- catch any column in input list that is in the list of table columns
```
Since this section should be refactored into an `inner join`, this comment should also move up.

Reply:
updated